Makefile

```diff
@@ -1,64 +1,57 @@
-
-include dependencies.makefile
+.PHONY: build install docker-image docker-image-build docker-image-ts1 docker-image-ts2-12 docker-image-ts2-13 docker-image-ts2-14 docker-image docker-image-build-2-12 docker-image-build-2-13 docker-image-build-2-14 docker-image-build-1
+.DEFAULT_GOAL := build
 
 PG_CONFIG?=pg_config
 
-EXTENSION=promscale
-
 EXT_VERSION = $(shell cat promscale.control | grep 'default' | sed "s/^.*'\(.*\)'$\/\1/g")
 PG_VERSION = $(shell ${PG_CONFIG} --version | awk -F'[ \.]' '{print $$2}')
 
-DATA = $(SQL_FILES)
-MODULE_big = $(EXTENSION)
-
-OBJS = $(SRCS:.c=.o)
-DEPS = $(SRCS:.c=.d)
-
-DEPS += target/release/libpromscale_rs.d
-
-SHLIB_LINK_INTERNAL = target/release/libpromscale_rs.a
-
-MKFILE_PATH := $(abspath $(MAKEFILE_LIST))
-CURRENT_DIR = $(dir $(MKFILE_PATH))
-
-TEST_PGPORT ?= 5432
-TEST_PGHOST ?= localhost
-TEST_PGUSER ?= postgres
-TESTS = $(sort $(wildcard test/sql/*.sql))
-USE_MODULE_DB=true
-REGRESS = $(patsubst test/sql/%.sql,%,$(TESTS))
-REGRESS_OPTS = \
-	--inputdir=test \
-	--outputdir=test \
-	--host=$(TEST_PGHOST) \
-	--port=$(TEST_PGPORT) \
-	--user=$(TEST_PGUSER) \
-	--load-language=plpgsql \
-	--load-extension=$(EXTENSION)
-
-PGXS := $(shell $(PG_CONFIG) --pgxs)
-
-EXTRA_CLEAN = $(DEPS)
-
-include $(PGXS)
-override CFLAGS += -DINCLUDE_PACKAGE_SUPPORT=0 -MMD
-override pg_regress_clean_files = test/results/ test/regression.diffs test/regression.out tmp_check/ log/
--include $(DEPS)
-
-all: target/release/libpromscale_rs.a
-
-rust: target/release/libpromscale_rs.a
-
-promscale.so: target/release/libpromscale_rs.a
-
-target/release/libpromscale_rs.a: Cargo.toml Cargo.lock $(RUST_SRCS)
+build:
 	cargo build --release --features pg${PG_VERSION} $(EXTRA_RUST_ARGS)
+	cargo pgx schema -f pg${PG_VERSION} --release
 
 clean:
-	rm -f $(OBJS) $(patsubst %.o,%.bc, $(OBJS))
-	rm -f promscale.so
 	cargo clean
 
-install: $(SQL_FILES)
-
-.PHONY: all docker-image docker-push rust
+package:
+	cargo pgx package --pg_config ${PG_CONFIG}
+
+install:
+	cargo pgx install --pg_config ${PG_CONFIG}
+
+PG_VER ?= pg12
+PUSH ?= FALSE
+DOCKER_IMAGE_NAME?=promscale-extension
+ORGANIZATION?=timescaledev
+
+docker-image-build-1 docker-image-build-2-12 docker-image-build-2-13 docker-image-build-2-14: Dockerfile $(SQL_FILES) $(SRCS) Cargo.toml Cargo.lock $(RUST_SRCS)
+	docker build --build-arg TIMESCALEDB_VERSION=$(TIMESCALEDB_VER) --build-arg PG_VERSION_TAG=$(PG_VER) -t $(ORGANIZATION)/$(DOCKER_IMAGE_NAME):$(EXT_VERSION)-$(TIMESCALEDB_VER)-$(PG_VER) .
```

Comment:
**Suggestion:** We probably want to use buildx for all of our image builds, since it has all of the rich features, including multi-arch images, which `docker build` doesn't support (currently anyway). It's just `docker buildx build` instead, which is basically identical with `docker build`, but also supports all of the extra functionality that buildx provides. It's especially important in order for caching work in GitHub Actions.

I maintain a handful of Docker images which use buildx and build/publish via GitHub Actions, so if you'd like some references to look at, just hit me up on Slack.

Reply:
https://github.com/timescale/promscale_extension/pull/52